source/plugin/scheduler.rst

```diff
@@ -163,3 +163,80 @@ In addition, there are a few other operations that are safe to do asynchronously
 
 * Independent network requests
 * Filesystem I/O (excluding files used by Sponge)
+
+Compatibility with other libraries
+==================================
+
+TODO: larger plugins mean you probably want to do some work asynchronously, note about standard java concurrency interfaces and thread-safety
+
+TODO: interfaces for both sync and async scheduling, sync is the most interesting.
+
+.. code-block:: java
+
+    import org.spongepowered.api.scheduler.SpongeExecutorService;
+    
+    SpongeExecutorService executor = Sponge.getScheduler().createSyncExecutor(plugin);
+    
+    // Execute a task on the primary server thread
+    executor.submit(() -> { ... });
+    // Execute a task on the primary server thread after 10 seconds
+    executor.schedule(() -> { ... }, 10, TimeUnit.SECONDS);
+
+TODO: nearly all large concurrency frameworks support some way of running using this interface
+
+CompletableFuture (Java 8)
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+TODO: difference between methods without and with *Async in their name
+
+.. code-block:: java
+
+    import java.util.concurrent.CompletableFuture;
+    
+    SpongeExecutorService executor = Sponge.getScheduler().createSyncExecutor(plugin);
+
+    CompletableFuture.supplyAsync(() -> {
+        // *Async methods default to running on ForkJoinPool.commonPool()
+        return /* awesome value */
+    }).thenAcceptAsync((awesomeValue) -> {
+        /* use awesomeValue on the main thread */
+    }, executor); // Run this future on our executor
+
+RxJava
```

Comment:
@kashike I dont think there is an example of external project integration in the documentation that I can base this section on, I presume adding a link to the title would break the documentation? :P